crates/bevy_pbr/src/light_probe/mod.rs

```diff
@@ -360,38 +384,114 @@ fn gather_light_probes<C>(
+                view_component,
+                &mut commands,
+                &mut reflection_probes,
+                &mut *view_reflection_probes,
+                &image_assets,
+            );
+        }
+
+        // Create the light probes for each face sub-camera of an
+        // omnidirectional camera, if necessary.
+
+        let (Some(cubemap_frusta), Some(omnidirectional_camera_entities)) =
+            (cubemap_frusta, omnidirectional_cameras.get(&view_entity))
+        else {
+            continue;
+        };
+
+        for (view_entity, frustum) in omnidirectional_camera_entities
+            .iter()
+            .zip(cubemap_frusta.iter())
+        {
+            gather_light_probes_for_camera(
+                *view_entity,
+                view_transform,
+                frustum,
+                maybe_render_layers,
+                view_component,
+                &mut commands,
+                &mut reflection_probes,
+                &mut *view_reflection_probes,
+                &image_assets,
+            );
+        }
+    }
+}
+
+/// Extracts light probes from the main world for a single camera.
+///
+/// [`gather_light_probes`] calls this both for standard 3D cameras and for the
+/// individual sub-cameras of each omnidirectional 3D camera.
+#[allow(clippy::too_many_arguments)]
+fn gather_light_probes_for_camera<C>(
+    view_entity: Entity,
+    view_transform: &GlobalTransform,
+    view_frustum: &Frustum,
+    maybe_view_render_layers: Option<&RenderLayers>,
+    view_component: Option<&C>,
+    commands: &mut Commands,
+    reflection_probes: &mut [LightProbeInfo<C>],
+    view_reflection_probes: &mut Vec<LightProbeInfo<C>>,
+    image_assets: &RenderAssets<GpuImage>,
+) where
+    C: LightProbeComponent,
+{
+    // Cull light probes outside the view frustum.
+    view_reflection_probes.clear();
+    view_reflection_probes.extend(reflection_probes.iter().filter_map(|light_probe_info| {
+        if !light_probe_info.frustum_cull(view_frustum) {
+            return None;
+        }
 
-        // Sort by distance to camera.
-        view_reflection_probes.sort_by_cached_key(|light_probe_info| {
-            light_probe_info.camera_distance_sort_key(view_transform)
-        });
-
-        // Create the light probes list.
-        let mut render_view_light_probes =
-            C::create_render_view_light_probes(view_component, &image_assets);
-
-        // Gather up the light probes in the list.
-        render_view_light_probes.maybe_gather_light_probes(&view_reflection_probes);
-
-        // Record the per-view light probes.
-        if render_view_light_probes.is_empty() {
-            commands
-                .get_or_spawn(view_entity)
-                .remove::<RenderViewLightProbes<C>>();
-        } else {
-            commands
-                .get_or_spawn(view_entity)
-                .insert(render_view_light_probes);
+        if let (Some(view_render_layers), Some(light_probe_render_layers)) = (
```

Comment:
In other places, we tend to fall back to default layer 0 if view layers are missing, is that intentionally not the case here?